cmd/modelcmd/base_test.go

```diff
@@ -407,3 +408,58 @@ func addCookie(c *gc.C, jar http.CookieJar, mac *macaroon.Macaroon, url *url.URL
 	cookie.Expires = time.Now().Add(time.Hour) // only persistent cookies are stored
 	jar.SetCookies(url, []*http.Cookie{cookie})
 }
+
+func (s *BaseCommandSuite) TestNewAPIRootWithOAuthLoginProvider(c *gc.C) {
```

Comment:
If we move the code to a function as suggested, that function can be tested in isolation and avoid all this.
We don't want to use reflect in tests - if we need access to unexported items we'd write the test in the same package as the code.

Reply:
done